cmake/Modules/MacroQbtCompilerSettings.cmake

```diff
@@ -73,20 +73,22 @@ macro(qbt_set_compiler_options)
         string(APPEND CMAKE_C_FLAGS " ${_GCC_COMMON_C_AND_CXX_FLAGS_STRING}")
         string(APPEND CMAKE_CXX_FLAGS " ${_GCC_COMMON_C_AND_CXX_FLAGS_STRING} ${_GCC_COMMON_CXX_FLAGS_STRING}")
 
-        # check whether we can enable -Og optimization for debug build
-        check_cxx_compiler_flag(-Og _DEBUG_OPTIMIZATION_LEVEL_IS_SUPPORTED)
+        if (CMAKE_BUILD_TYPE MATCHES "Debug")
+            # check whether we can enable -Og optimization for debug build
+            check_cxx_compiler_flag(-Og _DEBUG_OPTIMIZATION_LEVEL_IS_SUPPORTED)
 
-        if (_DEBUG_OPTIMIZATION_LEVEL_IS_SUPPORTED)
-            set(QBT_ADDITONAL_FLAGS "-Og -g3 -pipe" CACHE STRING
```

Comment:
From another viewpoint, is it important that we need to set these flags explicitly in debug mode? If not I would rather remove them. (Personally I think it is not)

Reply:
I don't mind removing them either

Reply:
I also think `-pipe` could be removed.

However, it would be best to have `-Og` in debug builds. From the gcc manpage:

```
If you are not using some other optimization option, consider using -Og with -g.
With no -O option at all, some compiler passes that collect information useful for debugging do not run at all, so that -Og may result in a better debugging experience.
```

and

```
-Og enables optimizations that do not interfere with debugging.
It should be the optimization level of choice for the standard edit-compile-debug cycle,
offering a reasonable level of optimization while maintaining fast compilation and a good debugging experience.
```

As for `-g3`, (the default level in gcc is `2` if only `-g` is specified):

```
Level 3 includes extra information, such as all the macro definitions present in the program.  Some debuggers support macro expansion when you use -g3.
```

Additional note: compatibility with other compilers could be a concern. For instance, `clang` has no `-g<level>` flags, it has different `-g`-family flags which override each other and do different things (`-gline-tables-only` and `-gmodules`), and in the `-g` section of its manpage it says: `Note that Clang debug information works best at -O0.` and `-Og Like -O1. In future versions, this option might disable different optimizations in order to improve debuggability.` (still true as of clang 10).

Reply:
>However, it would be best to have -Og in debug builds. 
>
>Additional note: compatibility with other compilers could be a concern. For instance, clang has no -g<level> flags

IMO, cmake itself should handle the compiler differences when debug mode is on and it is not qbt responsibility to babysit all the details. Also not to mention each developer would like to have their own debug flags and enforcing some debug flags by default is not a good idea.
Thus I'm still in favor of removing this whole section.

Reply:
> IMO, cmake itself should handle the compiler differences when debug mode is on and it is not qbt responsibility to babysit all the details (i.e. not related to qbt workflow in any way).

Agreed, but isn't there some due diligence that has to be done so that CMake knows it has to do this for you?

> Also not to mention each developer would like to have their own debug flags and enforcing some debug flags by default is not a good idea.

Somewhat disagree. I think it's ok to provide additional compiler options that are relevant to the project by default. It's just that care must be taken so that these flags don't clash with what `CMAKE_CXX_FLAGS_*` sets by default. Of course, the options should be easily customizable should a developer want to.

> Thus I'm still in favor of removing this whole section.

A rather extreme solution, but it there is no better one, I guess it works. I am very new to CMake myself, so I can only describe roughly what I would like to see, not how it is done.

Reply:
I think we don't need any option with cmake at all, cmake does fairly good job of managing compiler options across compilers, plus if you don't modify options in the cmake's code the compiler options are configurable by the cmake command line.

Reply:
IMO, only really required options should be hardcoded. E.g. qBittorrent needs debug info to be generated for crash reporting.

Reply:
Alright, based on the discussion here I have removed the block entirely. If we wished to keep the debug flags, instead of using additional flags, we could `CMAKE_C_FLAGS_DEBUG` and `CMAKE_CXX_FLAGS_DEBUG`.